aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1121,6 +1139,36 @@ static Tensor select_sparse(const Tensor& self, int64_t dim, int64_t index) {
   }
 }
 
+// this is an auxiliary function, called by the select&slice methods, that
+// creates a new quantizer from the given input
+QuantizerPtr create_subtensor_quantizer(const Tensor& self, bool is_select, int64_t start,
+  int64_t end, int64_t dim, int64_t step) {
+  auto quantizer_prev = get_qtensorimpl(self)->quantizer();
+  if (quantizer_prev->qscheme() == QScheme::PER_TENSOR_AFFINE) {
+    return quantizer_prev;
+  }
+  QuantizerPtr quantizer;
+  auto temp = static_cast<PerChannelAffineQuantizer*>(quantizer_prev.get());
+  auto axis = temp->axis();
+  auto scales = temp->scales();
+  auto zero_points = temp->zero_points();
+  if (dim == axis) {
+    // Compute scales&zps for sub-tensor
+    is_select ? scales = scales.select(0, start) : scales = scales.slice(0, start, end, step);
+    is_select ? zero_points = zero_points.select(0, start) : zero_points = zero_points.slice(0, start, end, step);
```

Comment:
can select be expressed with slice?

Reply:
also, probably clearer if we do:
```
scales = is_select ? ... : ...;
```
if we need to do this

Reply:
> can select be expressed with slice?

i think so, let me think. slice would maintain the dimension of the scales&zp to be 1 but select would just make it a scalar but that shouldn't affect the rest of the code